### PR TITLE
Revamp Query Editor

### LINKED ISSUE #124: Timestream Handover
As part of the handover of this plugin, we should:

 - [x] #130
 - [ ] #131

### PR DESCRIPTION
**Pull request by andresmgot**
Fixes #124 and #131

(sorry for the lengthy PR!)

Revamp the query editor to give a similar experience to other AWS SQL data sources:

Before:

Screenshot from 2022-02-21 17-07-56

After:

Screenshot from 2022-02-21 17-16-49

Summary of the changes:

 - Upgrade some dependencies and update tsconfig.
 - Remove frontend code to resolve databases, tables and measures (now resolved in the backend).
 - `SchemaInfo` class translated to `Suggestions` function.
 - Added tests.

### CODE REVIEW

File: cypress/integration/smoke.spec.ts [hunk 1]
```diff
@@ -49,12 +49,24 @@ export const addDataSourceWithKey = (datasourceType: string, datasource: any): a
...
+  };
 
   e2e.flows.addPanel({
     matchScreenshot: true,
     visualizationName: e2e.flows.VISUALIZATION_TABLE,
-    queriesForm: () => fillQuery(q),
+    queriesForm: () => {
+      fillQuery(q);
+      // Blur and wait
+      cy.get('.panel-content').last().click();
```
Comment (sarahzinger) [inline cypress/integration/smoke.spec.ts:66]: I follow the bits below, but can you talk through what this get panel-content is doing? Is it forcing a blur to trigger the query or something?

File: src/components/Suggestions.test.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+jest.mock('@grafana/runtime', () => {
+  return {
+    ...(jest.requireActual('@grafana/runtime') as any),
+    getTemplateSrv: () => templateSrv,
+  };
+});
+
+describe('getSuggestions', () => {
+  it('should include macros', () => {
```
Comment (sarahzinger) [inline src/components/Suggestions.test.tsx:18]: What would you think about adding under what circumstances macros should be included? I could see this test breaking in the future after a code change and it being hard to for the future dev to guess what this input represents for a user `[], [], [], [], '', '', ''`

**Review by sarahzinger — APPROVED** [review]
Cool stuff! I didn't do any manual testing but the code looks good to me :) I see in your screenshot that some items in the dropdowns are in quotes, what are your thoughts on removing them?

File: src/components/Suggestions.test.tsx (hunk 2 above)
Comment (andresmgot) [reply src/components/Suggestions.test.tsx:18]: true, these inputs are not very readable, I'm going to make these an object

**Comment by andresmgot** [comment]
> I see in your screenshot that some items in the dropdowns are in quotes, what are your thoughts on removing them?

I thought about removing the quotes when refactoring this plugin but didn't do it because it would be a small breaking change. I assume that there will be scenarios in which if you don't quote the database or the table, you would get an error. Also, in most of the examples in the docs (like [these](https://docs.aws.amazon.com/timestream/latest/developerguide/sample-queries.devops-scenarios.html)), the database is quoted (but I failed to find a reason why).

I could also remove the quotes from the selector and just use them in the resolved query but that could be confusing too. The measures are not being quoted so the behavior would be inconsistent.

**Pull request merged**